### DISCUSSION TITLE
ReactQueryDevtools z-index issue (and proposed solution)

### DISCUSSION THREAD

**Discussion by ykadosh** [TanStack/query#3775] [General]
The DevTools component (button & floating panel), despite having a very high `z-index`, can appear under elements that have a lower `z-index`, due to something called [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_Positioning/Understanding_z_index/The_stacking_context). In short, if one of your ancestors has a `z-index` lower than one of its siblings, then that sibling will be rendered over your element, regardless of your element's `z-index`.

The solution is to use a portal, and render the component in the body so that it is rendered over everything else regardless of the stacking context.

Has that been considered?

**Comment by TkDodo** [comment]
sounds good - happy to accept a PR 🙏

**Reply by ykadosh** [reply]
Happy to raise a PR 😁 
Can you kindly point me to the correct file in the repo?

**Reply by TkDodo** [reply]
start here: https://github.com/TanStack/query/tree/beta/packages/react-query-devtools
